avoidance/src/common.cpp

```diff
@@ -338,4 +338,38 @@ void fillUnusedTrajectoryPoint(mavros_msgs::PositionTarget& point) {
   point.yaw = NAN;
   point.yaw_rate = NAN;
 }
+
+// This function is a refactor of the original in the pcl library
+void removeNaNFromPointCloud(pcl::PointCloud<pcl::PointXYZ>& cloud, FOV& fov) {
+  // Filter out NANs and keep track of outermost points for FOV
+  size_t j = 0;
+  float h_max = -9999.f, v_max = -9999.f, h_min = 9999.f, v_min = 9999.f;
+  for (size_t i = 0; i < cloud.points.size(); ++i) {
+    if (!std::isfinite(cloud.points[i].x) ||
+        !std::isfinite(cloud.points[i].y) || !std::isfinite(cloud.points[i].z))
+      continue;
+    cloud.points[j] = cloud.points[i];  // safe, because i is always ahead of j
+    h_max = std::max(h_max, cloud.points[i].x / cloud.points[i].z);
```

Comment:
Divisions are really slow. I would instead suggest to check if the value is bigger than h_max via `h_max * z < x `, and then if so assign the division result.

Reply:
great idea! I put a good bit of thought into how to make this most efficient, but I hadn't thought of that. Main idea behind pulling this out of PCL was to avoid yet another loop through the entire point cloud